lib/content_location_filter.js

```diff
@@ -1,36 +1,39 @@
 "use strict";
 
 const HyperSwitch = require('hyperswitch');
-const Template = HyperSwitch.Template;
 const URI = HyperSwitch.URI;
 const mwUtil = require('./mwUtil');
 
-let basePathTemplate;
-
-module.exports = (hyper, req, next, options) => {
-    basePathTemplate = basePathTemplate || new Template({
-        uri: options.templates.base_uri_template
-    });
+module.exports = (hyper, req, next) => {
     if (req.method !== 'get') {
         return next(hyper, req);
     } else {
-        const baseUri = basePathTemplate.expand({
-            request: req
-        }).uri;
         const attachLocation = (res) => {
-            if (res.status !== 301 && res.status !== 302) {
-                res.headers = res.headers || {};
+            res.headers = res.headers || {};
+            if (res.status === 301 || res.status === 302
+                    || res.headers['content-location']) {
+                return res;
+            }
+            return hyper.get({
+                uri: new URI([req.params.domain, 'sys', 'action', 'siteinfo'])
```

Comment:
+1 on that. That method could also extract the body of the response right away to be compatible with the current code.